hypothesis-python/src/hypothesis/internal/conjecture/junkdrawer.py

```diff
@@ -101,9 +115,14 @@ def append(self, n):
                 return
             except OverflowError:
                 assert n > 0
-                self.__underlying = array_or_list(
-                    NEXT_ARRAY_CODE[self.__underlying.typecode], self.__underlying
-                )
+                self.__upgrade()
+
+    def __upgrade(self):
+        code = NEXT_ARRAY_CODE[self.__underlying.typecode]
+        if code == "O":
```

Comment:
This branch is redundant, given the changes to `array_or_list` to handle object arrays.